flashcards/web/src/components/AddDecksForm.tsx

```diff
@@ -0,0 +1,160 @@
+// @ts-nocheck
+
+import React from 'react';
+import {useFormik} from 'formik';
+import {
+    Button, Checkbox,
+    FormControlLabel,
+    Grid,
+    InputLabel,
+    MenuItem,
+    Select,
+    TextField,
+    Typography
+} from "@mui/material";
+import * as Yup from "yup";
+
+const AddDecksForm = () => {
+    const formik = useFormik({
+        validateOnChange: false,
+        initialValues: {
+            tagName: '',
+            deckName: '',
+            selectTag: '',
+            selectCategory: '',
+            selectAvailability: '',
+            price: '',
+            checkboxPublic: false,
+            checkboxPrice: false
+        },
+        onSubmit: values => {
+            alert(JSON.stringify(values, null, 2));
+        },
+        validationSchema: Yup.object().shape({
+            tagName: Yup.string()
+                .min(2, 'Too Short!')
+                .max(20, 'Too Long!')
+                .required('Required!'),
+            deckName: Yup.string()
+                .min(2, 'Too Short!')
+                .max(20, 'Too Long!')
+                .required('Required!'),
+            checkboxPrice: Yup.boolean(),
+            checkboxPublic: Yup.boolean(),
+            price: Yup.number()
+                .when('checkboxPrice', {
+                    is: true,
+                    then: Yup.number().required('Required!')
+                }),
+            selectTag: Yup.string()
+                .required('Required!'),
+            selectCategory: Yup.string()
+                .required('Required!')
+        })
+    });
+    return (
+        <form onSubmit={formik.handleSubmit}>
+            <Grid container maxWidth="md">
```

Comment:
Try using Stack instead of bunch of grid, It's not really a grid it's a column.

Reply:
I missed this message, is the current version of the form acceptable?